### ISSUE TITLE
Rubocop Autocorrect Produces Buggy Code (syntax error)

### ISSUE THREAD

**Issue by jonmchan** [bbatsov/rubocop#4306]
Rubocop autocorrect produced buggy code with syntax error.  It added a )( in an odd position (```[role)](&&```)

--------

## Expected behavior

Autocorrect should correct with the following:
```
   def is_authorized?(role = 'viewer')
     return true if isAdmin?
     enabled? && AuthUser.roles[role] && AuthUser.roles[self.role] == AuthUser.roles[role]
   end
```

## Actual behavior

Autocorrect made the following changes:

```
   def is_authorized?(role = 'viewer')
-    return true if isAdmin?
-    enabled? and AuthUser.roles[role] and AuthUser.roles[self.role] == AuthUser.roles[role]
+    return true if isAdmin?
+    enabled? && AuthUser.roles[role)](&& AuthUser.roles[self.role] ==(AuthUser.roles[role])
   end
```

## Steps to reproduce the problem

Create a class with the following function:
```
class AuthUser
   def is_authorized?(role = 'viewer')
    return true if isAdmin?
    enabled? and AuthUser.roles[role] and AuthUser.roles[self.role] == AuthUser.roles[role]
   end
end
```

Run ```bundle exec rubocop -a```.

## RuboCop version

```
$ bundle exec rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-linux)
```

**Comment by lassik** [comment]
Here's another case. `attempted.rb`:

    current_user = { is_admin: true }
    admin_data = if current_user and current_user[:is_admin]
                   {
                     users: 'foo'
                   }
                 end

Output of `rubocop -a attempted.rb`:

    attempted.rb:2:30: C: [Corrected] Use && instead of and.
    admin_data = if current_user and current_user[:is_admin]
                                 ^^^
    attempted.rb:2:55: E: unexpected token tRPAREN
    (Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
    admin_data = if current_user && current_user[:is_admin)](               {
                                                          ^
    attempted.rb:5:14: E: unexpected token kEND
    (Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
                 end
                 ^^^

Is #3475 related?

**Comment by Drenmi** [comment]
Will possibly be resolved by #4381. 🙂

**Comment by rrosenblum** [comment]
I just tested this against master. It does appear to have been fixed.
